### ISSUE TITLE
MACOSX_DEPLOYMENT_TARGET

### ISSUE THREAD

**Issue by trevorsandy** [devernay/osmesa-install#2]
Would it make sense to add these directives to the script so one can suppress the numerous warnings when building with clang ?

`MACOSX_DEPLOYMENT_TARGET=10.<X>` `-mmacosx-version-min=10.<X>` `-isysroot` `/Developer/SDKs/MacOSX10.<X>.sdk`

Just to add a little background...

If I'm building with Qt (QtCreator), Qt clang spec will automatically set my macosx-version-min to SDK 10.8. So if I build my osmesa libs on my current version of OSX (SDK 10.12) using osmesa-install as is and then use the libs in my application build, I'll generate numerous warnings that the SDK for the compiled version of of the libs is 10.12 which is different from the Qt compiled application at SDK 10.8.
Cheers,

**Comment by devernay** [comment]
did you simply try setting de env variable `MACOSX_DEPLOYMENT_TARGET` to 10.8 when compiling osmesa?
as in `env MACOSX_DEPLOYMENT_TARGET=10.8 ../osmesa-install.sh`

that may do the trick.

Are you using Qt 4 or Qt 5?

Are you using the `unsupported/macx-clang` spec? Most Qt installations (eg from MacPorts and homebrew) properly patch the default config (`macx-g++` or `macx-xcode`), so the default config should be used instead.

**Comment by trevorsandy** [comment]
Actually, I followed your guidance from  Xcodelegacy which favoured using the c/cxxflags directive over the environment variable. As stated in the Xcodelegacy readme, the env var is suggested to support old compilers. 

In fact, the updated version of the script (I've created a pull request so you can review) does exactly what you suggest above plus formally adds the full directive to cmake (for llvm) and c/cxxflags for osmesa's configure.

I'm using the Qt distro (5.7.1) from Qt which has its own installer and update framework. I also build statically from source. 

I'm using the macx-clang spec. I did not know this spec was unsupported because it's the default spec automatically selected by QMake when generating makefiles.

Out of curiosity, what would be the SDK compatibility behaviour using a 'supported' Qt spec ? I ask because, I don't see a bug in the current spec's behaviour. The Qt/QMake macx-clang spec, and its associated conf includes, automatically sets the OSX SDK deployment target to 10.8. 

Using clang directly does not set a target because, it expects you to do so as a directive. When, I'm building something outside of Qt/Xcode, like osmesa, or llvm for example, I expect to add the directive defining which SDX I'm targeting for backward compatibility. It was this expectation that prompted me to add the directive to your osmesa-install script. 

Cheers,

**Comment by devernay** [comment]
macx-clang was unsupported in 4.8, it is supported in Qt5.
I merged your changes. Please avoid reformatting in your pull requests (or do a pull request with only format changes), because it makes reviewing much more difficult!

**Comment by devernay** [comment]
I have to modify the script so that if you want to set macosx-version-min, you have to set the `MACOSX_DEPLOYMENT_TARGET` env var, as in:
```
env MACOSX_DEPLOYMENT_TARGET=10.8 ../osmesa-install.sh
```

**Comment by devernay** [comment]
same for the SDK root, it uses the SDKROOT environment variable, which is the standard name.
See:
https://cmake.org/cmake/help/v3.0/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html
https://cmake.org/cmake/help/v3.0/variable/CMAKE_OSX_SYSROOT.html

**Comment by trevorsandy** [comment]
Very good. Much cleaner! This is what I wanted to do but as I'm not so deep in MacOSX platform development and I wasn't sure of your script design approach for OSX versions older that 10, I didn't want to step on those parts of the script.

On another note. I've made allot of useful enhancements on my version as I continue to have problems building on MinGW/Scons (see [this gist entry](https://gist.github.com/trevorsandy/69d22f8a0ceeafe298baba9587cd37e9) I generated from the script). I'm not sure if it's an incomplete environment issue or if there is indeed a bug in the Windows source. I've fulfilled all the stated environment requirements but I still get a consistent 'expected initializer ...' failure.

OSX, and Linux (Ubuntu) works very well - even Windows Subsystem for Linux (WSL) which also uses Ubuntu works well. 

I'll update to 17.1.3 to see if any improvement on the MinGW build. I'll send a pull request after. Just cancel if you are not interested in the updates.
